### PR TITLE
LSE max wait patches for rusEFI

### PR DESCRIPTION
**Pull request by nmschulte**
for https://github.com/rusefi/rusefi/issues/4556, with https://github.com/rusefi/rusefi/pull/4944

### CODE REVIEW

File: os/hal/ports/STM32/STM32F4xx/hal_lld.c [hunk 1]
```diff
@@ -64,15 +64,17 @@ static void hal_lld_backup_domain_init(void) {
...
 #if defined(STM32_LSE_BYPASS)
   /* LSE Bypass.*/
   RCC->BDCR |= RCC_BDCR_LSEON | RCC_BDCR_LSEBYP;
 #else
   /* No LSE Bypass.*/
   RCC->BDCR |= RCC_BDCR_LSEON;
 #endif
-  while ((RCC->BDCR & RCC_BDCR_LSERDY) == 0)
-    ;                                       /* Waits until LSE is stable.   */
+  while (rusefiLseCounter++ < RUSEFI_STM32_LSE_WAIT_MAX
```
Comment (mck1117) [inline os/hal/ports/STM32/STM32F4xx/hal_lld.c:75]: do we want to change the value of RTCSEL based on whether or not HSE works?

Comment (nmschulte) [reply os/hal/ports/STM32/STM32F4xx/hal_lld.c:75]: I assume you mean LSE (vs HSE)?

Reviewing the code, it seems like that's important, however the logic in ChibiOS 16 patch doesn't do this: I guess rusEFI is changing the RTC source already, or this oversight is part of the/a bug that's existed then.

Comment (mck1117) [reply os/hal/ports/STM32/STM32F4xx/hal_lld.c:75]: sorry I mean LSE - right now it'll use the value of RTCSEL blindly even if LSE doesn't work, when it should instead use LSI

**Comment by rusefillc** [comment]
This PR has suddenly went south with 3200 files changed?

**Comment by nmschulte** [comment]
> This PR has suddenly went south with 3200 files changed?

I was accidentally basing `master` whereas rusEFI targets `stable_20.3.x.rusefi`; fixed.

File: .gitattributes [hunk 2]
```diff
@@ -0,0 +1 @@
+* text=auto
```
Comment (nmschulte) [inline .gitattributes:1]: @rusefillc ChibiOS `stable_20.3.x.rusefi` uses CRLF line-endings -- I made my changes against `master` which had LF line-endings ... this addition (and `git add --renormalize os/hal/ports/STM32` made fixing my rebase easier -- is this okay or should we avoid this, or perhaps renormalize the entire tree?

Comment (mck1117) [reply .gitattributes:1]: why is your git line ending mode set to the mode where it doesn't change to your machine's culture?

Comment (nmschulte) [reply .gitattributes:1]: It is setup with auto/native, but for some reason `rebase` wasn't having it with the difference and generating full-file 3-ways.  I'm happy to remove this and retain CRLF in the repository, but I believe that's the culprit (the committed files are CRLF, and `rebase` wasn't detecting them as text).

Comment (nmschulte) [reply .gitattributes:1]: Oh shit, `core.autocrlf` was in fact commented out in my config. :+1:

File: os/hal/ports/STM32/STM32F4xx/hal_lld.c (hunk 1 above)
Comment (nmschulte) [reply os/hal/ports/STM32/STM32F4xx/hal_lld.c]: I implemented a fallback as it seems necessary.  I believe without it the CPU resets, but unsure (all LEDs other than Power dimly lit).

Comment (mck1117) [reply os/hal/ports/STM32/STM32F4xx/hal_lld.c]: It doesn't reset, it just hangs during init of some step that relies on the oscillator being there, which it isn't.

Comment (nmschulte) [reply os/hal/ports/STM32/STM32F4xx/hal_lld.c]: Ah okay.  Seems the mode is the same as without this patch (but STM32_LSE_ENABLED; waiting in the busy-loop).  I wonder what causes the LEDs to be dim, but no matter.

Comment (mck1117) [reply os/hal/ports/STM32/STM32F4xx/hal_lld.c]: with a debugger you can just break in and see where it's hung

Comment (nmschulte) [reply os/hal/ports/STM32/STM32F4xx/hal_lld.c]: Debugger comes next week :+1:.

**Review by mck1117 — APPROVED** [review]

**Pull request merged**